### DISCUSSION TITLE
Fish isn't picking `$XDG_CONFIG_HOME` when logging in

### DISCUSSION THREAD

**Discussion by Samueru-sama** [fish-shell/fish-shell#10739] [Q&A]
I'm a zsh user that has `export ZDOTDIR="$HOME/Local/config/zsh"` in `etc/zshenv` and when I login my .zprofile there is sourced whch sets the rest of env variables, etc, etc. One of those is `XDG_CONFIG_HOME="$HOME/Local/.config"`

Since fish doesn't have something like ZDOTDIR (not sure) and instead follows the XDG Base dir spec already, I have to first set `$XDG_CONFIG_HOME` before I log in.

I tried adding `export XDG_CONFIG_HOME=$HOME/Local/config` to `/etc/profile`, but it didn't work. I also tried adding `set -Ux XDG_CONFIG_HOME $HOME/Local/config` to `/etc/fish/config.fish` and didn't work either.

Fish isn't picking XDG_CONFIG_HOME and creates the default directories `~/.config/fish` and `~/.local/share/fish` when I log in.

I'm not sure if what I'm trying to do is even possible, since when I search for solutions to get rid of all the dotfiles in the top level of home what is always suggested is to use zsh and set ZDOTDIR.

**Comment by faho** [comment]
> I also tried adding set -Ux XDG_CONFIG_HOME $HOME/Local/config to /etc/fish/config.fish 

Any fish file or place you add it to will be too late because it is *after* fish has already determined the files to read. Universal variables in particular are stored in $XDG_CONFIG_HOME/fish/fish_variables.

It needs to be set *before* fish, so that fish picks it up as soon as it starts.

> I tried adding export XDG_CONFIG_HOME=$HOME/Local/config to /etc/profile

This would work if a shell that reads /etc/profile is your login shell. Fish doesn't, so if it is, that file is never executed.

-----

Frankly, my recommendation is to just not bother and use ~/.config.

**Comment by Samueru-sama** [comment]
@faho I was able to fix the issue by defining the vars in `/etc/enviroment`

```
XDG_DATA_HOME=/home/samuel/Local/share
XDG_CONFIG_HOME=/home/samuel/Local/config
```

And yes it has to be full path since it seems I can't have variables in `/etc/enviroment`.

And I had to define both `XDG_CONFIG_HOME` and `XDG_DATA_HOME`, because with only `XDG_CONFIG_HOME` it resulted in fish still making the `~/.local/share/fish` directory upon login.

I didn't like that with zsh I had to set ZDOTDIR for all users so I was hoping that with fish I wouldn't have this problem, but the only solution I've found is even worse.
